### PR TITLE
Fix deleting model definition from not main connection

### LINKED ISSUE #118: deleteModel try to remove model only from first/main mongoose connection
## Versions

- Typegoose(NPM): 6.1.0 

## What is the Problem?
method `.deleteModel()` try to remove model only from first/main mongoose connection
this issue is blocking -> https://github.com/kpfromer/nestjs-typegoose/issues/41 

## Do you know *why* it happenes?
https://github.com/typegoose/typegoose/blob/a3f068ed58713adee017c39e21011e934061e987/src/typegoose.ts#L225
`mongoose.connection` - it's the first and default connection in `mongoose`. And if you create another connection then *deleting this model* would not work properly because there will not be such model in first connection.

To soleve this problem you should find this model in all connections:

```ts
const connection  = mongoose.connections.find((con) => con.models.has(modelForRemove));
if (connection) {
  connection.deleteModel(modelForRemove);
}
```
Something like this.
p.s. Haven't checked this code😅

### PR DESCRIPTION
**Pull request by Goodluckhf**
## Misc

- [ ] Is this a prototype? 
- [x] Written Tests for it? 
- [ ] Already read & followed [CONTRIBUTING](https://github.com/typegoose/typegoose/tree/master/.github/CONTRIBUTING.md)?

## Related Issues

- closes #118

### CODE REVIEW

File: test/utils/mongooseConnect.ts [hunk 1]
```diff
@@ -12,18 +12,35 @@ if (config.Memory) {
...
+
 /**
  * Make a Connection to MongoDB
  */
-export async function connect(): Promise<void> {
+export async function connect(extraConfig: ExtraConnectionConfig = {}): Promise<mongoose.Connection> {
   if (config.Memory) {
-    await mongoose.connect(await instance.getConnectionString(), {
+    if (extraConfig.createNewConnection) {
+      return mongoose.createConnection(await instance.getConnectionString(extraConfig.dbName), {
```
Comment (hasezoey) [inline test/utils/mongooseConnect.ts]: could you make the same for not in memory db?

Comment (Goodluckhf) [reply test/utils/mongooseConnect.ts]: yep, done!

**Comment by Goodluckhf** [comment]
> i will test this locally, but one thing i noticed

You can run tests with my commit [test case](8a10db3db3f52add6022c1fc5e502033fcfce308)

**Comment by hasezoey** [comment]
> You can run tests with my commit test case

i dont understand what you mean with this

PS: your link is broken (404)

**Comment by Goodluckhf** [comment]
yea, sorry. I've fix the link

**Comment by hasezoey** [comment]
@Goodluckhf are you ok with the changes i made?

**Comment by Goodluckhf** [comment]
Yeah that's all right

**Comment by hasezoey** [comment]
Thanks for pointing this out again, i missed this because we dont have tests for `existingMongoose` / `existingConnection` yet

**Pull request merged**